### PR TITLE
feat: add button example and fix button icon binding.

### PR DESCRIPTION
**Pull request by Trancever**
Icon prop bindings were wrong. In paper2.0 we no longer support passing ReactElements as prop. The right way is to use render prop right now.

### CODE REVIEW

**Review by medson10 — COMMENTED** [review]
Fix git conflicts

File: example/src/Home.re [hunk 1]
```diff
@@ -18,10 +18,21 @@ type screen = {
 type examples = array(screen);
 
 let examples = [|
+<<<<<<< Updated upstream
```
Comment (medson10) [inline example/src/Home.re]: here

File: example/src/Navigation.re [hunk 2]
```diff
@@ -4,7 +4,15 @@ module Config = {
     | SnackbarExample
     | SurfaceExample
     | TypographyExample
+<<<<<<< Updated upstream
```
Comment (medson10) [inline example/src/Navigation.re]: and here

**Review by medson10 — APPROVED** [review]

**Comment by Trancever** [comment]
Thanks for review! 👍

**Pull request merged**